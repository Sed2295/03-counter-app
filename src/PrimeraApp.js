import React from 'react';
//import React,{Fragment} from 'react';
import PropTypes from 'prop-types';
const PrimeraApp = ({ saludo,subtitulo }) => {
    /* const saludo = {
        nombre:'Alfredo',
        edad: 25
    } */
    /* if(!saludo)
        throw new Error('El saludo es necesario') */
    return (
        <>
            <h1>{saludo}</h1>
            {/*<pre>{ JSON.stringify(saludo,null,3) }</pre>*/}
            <p>Mi primer parrafo</p>
        </>        
    ); 
}
//Obligamos que cuando se necesite utilizar el componente venga con el tipado que necesitamos esto con propTypes
PrimeraApp.propTypes = {
    saludo:PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo : 'Soy un subtitulo',

}
export default PrimeraApp;