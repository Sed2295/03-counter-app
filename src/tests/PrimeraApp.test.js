import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
//import {render} from '@testing-library/react'; para jest
import PrimeraApp from '../PrimeraApp';


describe('Pruebas en primera App', () => {
   /* codigo usando pruebas con jest   
    test('debe mostrar el mensaje hola soy goku', () => {
        const saludo = 'Hola soy Goku';
        const {getByText} = render(<PrimeraApp saludo = {saludo}/>);
        expect ( getByText(saludo) ).toBeInTheDocument();
    })   */
    test('debe de mostrar <PrimeraApp>correctamente', () => {
        const saludo = 'Hola soy goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    })
    test('debe enviar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy goku',
            subtitulo = 'Mi primer parrafo';
        const wrapper = shallow(<PrimeraApp saludo = {saludo} subtitulo={subtitulo}/>)
        //buscamos en el dom la etiqueta p, wrapper nos trae todo lo de <PrimeraAPP/>
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);
    })
    
       
})
//en consola presionamos u para actualizar un snapshot
