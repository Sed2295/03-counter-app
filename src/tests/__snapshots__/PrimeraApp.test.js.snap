// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera App debe de mostrar <PrimeraApp>correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy goku
  </h1>
  <p>
    Mi primer parrafo
  </p>
</Fragment>
`;
